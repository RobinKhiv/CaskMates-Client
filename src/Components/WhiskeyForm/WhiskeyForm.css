.newWhiskey-form{
    margin-top: 10px;
    display: flex;
    flex-direction: column;
    border-radius: 3px; 
    background-color: rgba(233, 219, 204, 0.788);
    padding-bottom: 10px;
    border: 1px solid #7220021f;
    margin-bottom: 7px;
}
.whiskeyFormTitle{
    text-align: center;
    margin-bottom: 4px;
    margin-top: 4px;
}

#submit-btn{
    margin-top: 3px;
    margin-left: auto;
    margin-right: auto;
}
@media only screen and (max-width: 640px) {
    .newWhiskey-form.col-4{
        width: 80%;
        padding-left: 5px;
        padding-right: 5px;
    }
}