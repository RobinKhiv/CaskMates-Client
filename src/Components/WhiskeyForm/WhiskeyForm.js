import React, { Component } from 'react'
import whiskeysService from '../../Services/whiskey-api-service';
import './WhiskeyForm.css'

export default class WhiskeyForm extends Component {
  onSubmit = e => {
      e.preventDefault();
      const {drinkName, drinkImage, origin, abv, price, content, nose, palate, finish} = e.target;
      whiskeysService.postWhiskey(
        drinkName.value, 
        drinkImage.value, 
        origin.value, 
        parseFloat(abv.value), 
        parseFloat(price.value), 
        content.value, 
        nose.value, 
        palate.value, 
        finish.value)
      .then(whiskeyId => this.props.onAddSuccess(whiskeyId))
  }
  
  render() {

    return (
      <form onSubmit={this.onSubmit} className="newWhiskey-form col-6">
        <h1 className="whiskeyFormTitle">Add a Drink</h1>
        <label htmlFor="drinkName">Drink: </label>
        <input type="text" name="drinkName" id="drinkname"className='wide'/><br/>
        <label htmlFor='drinkImage'>Image Link: </label>
        <input type="text" name="drinkImage" id="drinkImage"/><br/>
        <label htmlFor="origin">Origin: </label>
        <input type="text" name="origin" id="origin"/><br/>
        <label htmlFor="abv">Abv: </label>
        <input type="text" name="abv" id="abv"/><br/>
        <label htmlFor="price">Price: </label>
        <input type="text" name="price" id="price"/><br/>
        <label htmlFor="content">Description: </label>
        <input type="text" name="content" id="content"/><br/>
        <label htmlFor="nose">Nose: </label>
        <input type="text" name="nose" id="nose"/><br/>
        <label htmlFor="palate">Palate: </label>
        <input type="text" name="palate" id="palate"/><br/>
        <label htmlFor="finish">Finish: </label>
        <input type="text" name="finish" id="finish"/><br/>
        <input type="submit" name="submit" id="submit-btn" value="submit"/>
        <input type="button" value="Cancel" id="cancel-btn"></input>
      </form> 
    )
  }
}
