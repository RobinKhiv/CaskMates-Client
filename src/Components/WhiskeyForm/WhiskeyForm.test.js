import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import WhiskeyForm from './WhiskeyForm'

describe('<WhiskeyForm />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render( <WhiskeyForm /> , div);
        ReactDOM.unmountComponentAtNode(div);
    });

  it('renders without crashing', () => {
    const tree = renderer
      .create(<WhiskeyForm />)
      .toJSON();
    expect(tree).toMatchSnapshot(); 
  });
});