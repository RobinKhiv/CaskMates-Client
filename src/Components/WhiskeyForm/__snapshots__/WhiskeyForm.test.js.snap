// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhiskeyForm /> renders without crashing 1`] = `
<form
  className="newWhiskey-form col-6"
  onSubmit={[Function]}
>
  <h1
    className="whiskeyFormTitle"
  >
    Add a Drink
  </h1>
  <label
    htmlFor="drinkName"
  >
    Drink: 
  </label>
  <input
    className="wide"
    id="drinkname"
    name="drinkName"
    type="text"
  />
  <br />
  <label
    htmlFor="drinkImage"
  >
    Image Link: 
  </label>
  <input
    id="drinkImage"
    name="drinkImage"
    type="text"
  />
  <br />
  <label
    htmlFor="origin"
  >
    Origin: 
  </label>
  <input
    id="origin"
    name="origin"
    type="text"
  />
  <br />
  <label
    htmlFor="abv"
  >
    Abv: 
  </label>
  <input
    id="abv"
    name="abv"
    type="text"
  />
  <br />
  <label
    htmlFor="price"
  >
    Price: 
  </label>
  <input
    id="price"
    name="price"
    type="text"
  />
  <br />
  <label
    htmlFor="content"
  >
    Description: 
  </label>
  <input
    id="content"
    name="content"
    type="text"
  />
  <br />
  <label
    htmlFor="nose"
  >
    Nose: 
  </label>
  <input
    id="nose"
    name="nose"
    type="text"
  />
  <br />
  <label
    htmlFor="palate"
  >
    Palate: 
  </label>
  <input
    id="palate"
    name="palate"
    type="text"
  />
  <br />
  <label
    htmlFor="finish"
  >
    Finish: 
  </label>
  <input
    id="finish"
    name="finish"
    type="text"
  />
  <br />
  <input
    id="submit-btn"
    name="submit"
    type="submit"
    value="submit"
  />
  <input
    id="cancel-btn"
    type="button"
    value="Cancel"
  />
</form>
`;
